src/Comps/search/Search.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ChangeEvent, useState ,useEffect} from 'react';
+import { SearchIcon } from "../../Icons";
+import {
+  SearchFormContainer,
+  SearchButton,
+  Input,
+  VerticalLine,
+} from "./style";
+
+// import DropDown from '../dropdown/DropDown';
+
+function useDebounce<T>(value: T, delay?: number): T {
```

Comment:
Remember always to think more generally and to make separation between ui and logic.
Think what would happen if you'd have another place in the system that you need to use the search bar and need a different debounce time or don't need debounce at all

Reply:
Ok thank you.
working on it